### DISCUSSION TITLE
cad model of NREL blade and adding my own blade to openfast

### DISCUSSION THREAD

**Discussion by lt72884** [OpenFAST/openfast#2011] [Q&A]
Hello all:) thank you for the previous help i have recieved. i am very grateful for it.

First item of business, I am trying to find a good CAD model of the blade that is used for the 5mw land dll windturb openfast simulations. I would like to practice my FEA but i can not find a decent model anywhere. STEP file is much appreciated, if one exists.

Second, i have modeled my own 650mm wind turbine blade in freeCAD based on the A 18 original airfoil. Is it possible to add this blade, its mass, and density to openfast to run simulations on my blade? If not, do you know of any opensource software that will allow me to add my blade and produce similar results as what openfast does? 

thanks

**Comment by jjonkman** [comment]
Dear @lt72884,

Regarding a CAD model of the NREL 5-MW baseline wind turbine blade, see my response to a similar question asked and answered on our forum: https://forums.nrel.gov/t/nrel-5-mw-blade-airfoils-data-files-ar17/5561/7?u=jason.jonkman.

OpenFAST is fully parameterizeable, and so, you should be able to define any blade in OpenFAST.  Of course, instead of the CAD geometry, OpenFAST will expect the inputs to be defined in terms beam-type structural properties (cross-sectional mass and stiffness, etc.) in ElastoDyn or BeamDyn and actuator-line-type aerodynamic properties (chord, twist, airfoil polar data, etc.) in AeroDyn.

Best regards

**Comment by lt72884** [comment]
ok, man, that sounds intemidating to do haha. So i need to edit the elastodyn or beamdyn file as well as the aerodyn file. 

If i have those parameters, if i open one of those files in notepad++, what am i changing? is there a guide on this particular aspect of openfast?

As for the cad files, thank you for providing that link. It helps alot. I could model it, but wanted to check with you first to see if one was already created.

thanks

**Reply by jjonkman** [reply]
Dear @lt72884,

Regarding editing the ElastoDyn, BeamDyn, and AeroDyn input files, etc., I would suggest reviewing the documentation of each module to understand what input parameters need to be set:
https://openfast.readthedocs.io/en/main/source/user/elastodyn/index.html
https://openfast.readthedocs.io/en/main/source/user/beamdyn/index.html
https://openfast.readthedocs.io/en/main/source/user/aerodyn/index.html

Best regards,

**Comment by lt72884** [comment]
ok, thank you for posting this : )
